# Life Expectancy Analysis & Modeling Using WHO, UN data

## Linear Model Coefficient Investigation

To:&nbsp;&nbsp;&nbsp;&nbsp; [Magnimind](https://magnimindacademy.com/)

From: Matt Curcio, matt.curcio.ri@gmail.com

Date: 2023-01-29

Re:&nbsp;&nbsp;&nbsp; NOTEBOOK #8

---


In [7]:
# Common Python Libraries
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
%matplotlib inline
import seaborn as sns

# import warnings
import warnings
warnings.filterwarnings("ignore")

# Loading Modeling Libaries
from sklearn.linear_model import LinearRegression
from sklearn import linear_model
from sklearn.model_selection import train_test_split, cross_val_score

In [8]:
path = '../data/processed/'
fileName = 'Clean_LE_Data_FEng_4.csv'

df = pd.read_csv(path+fileName, header=0)

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2928 entries, 0 to 2927
Data columns (total 18 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   Country         2928 non-null   object 
 1   Year            2928 non-null   int64  
 2   Status          2928 non-null   int64  
 3   LifeExpectancy  2928 non-null   float64
 4   AdultMort       2928 non-null   float64
 5   EtOH            2928 non-null   float64
 6   PercExpen       2928 non-null   float64
 7   Measles         2928 non-null   int64  
 8   BMI             2928 non-null   float64
 9   lt5yD           2928 non-null   int64  
 10  Polio           2928 non-null   float64
 11  TotalExpen      2928 non-null   float64
 12  DTP             2928 non-null   float64
 13  HIV             2928 non-null   float64
 14  Thin1_19y       2928 non-null   float64
 15  Income          2928 non-null   float64
 16  Education       2928 non-null   float64
 17  Region          2928 non-null   i

In [10]:
# Data splits

y = df['LifeExpectancy']

x = df[['Income','Education','Polio','DTP','AdultMort','HIV']]

x_train, x_test, y_train, y_test = train_test_split(x, y, test_size = 0.3, random_state = 100)

## Linear Regression Parameter Investigation

In [11]:
reg_model = linear_model.LinearRegression()

reg_model = LinearRegression().fit(x_train, y_train)

#Printing the model coefficients
print('Intercept: ',reg_model.intercept_)

# pair the feature names with the coefficients
list(zip(x, reg_model.coef_))

Intercept:  50.66478178594173


[('Income', 9.113970547007359),
 ('Education', 0.8860513547432141),
 ('Polio', 0.037639530276743494),
 ('DTP', 0.04333747425602766),
 ('AdultMort', -0.02244631132206054),
 ('HIV', -0.47927856071777236)]

$$Life~Expectancy = 50.7 \\+ ~9.1 \cdot Income \\+ 0.90 \cdot Education \\+ 0.04 \cdot Polio \\+ 0.04 \cdot DTP \\- 0.02 \cdot Adult~Mortality \\- 0.48 \cdot HIV$$